app/controllers/api/discussions_controller.rb

```diff
@@ -3,60 +3,52 @@ class API::DiscussionsController < API::RestfulController
   load_resource only: [:create, :update]
 
   def dashboard
-    instantiate_collection { |collection| filter_collection collection }
-    respond_with_discussions
+    instantiate_collection { |collection| collection_for_dashboard collection }
+    respond_with_collection scope: { visible_in_dashboard: true }, serializer: DiscussionWrapperSerializer, root: 'discussion_wrappers'
+  end
+
+  def inbox
+    instantiate_collection { |collection| collection_for_inbox collection }
+    respond_with_collection scope: { visible_in_inbox: true }, serializer: DiscussionWrapperSerializer, root: 'discussion_wrappers'
```

Comment:
Yep, otherwise you get an awkward half-loading screen where it looks like
you only have unread threads in your recent threads list until the rest of
the request comes down.

On Mon, Aug 3, 2015 at 11:56 AM, Robert Guthrie <notifications@github.com>
wrote:

> In app/controllers/api/discussions_controller.rb
> <https://github.com/loomio/loomio/pull/2369#discussion_r36117990>:
>
> > @@ -3,60 +3,52 @@ class API::DiscussionsController < API::RestfulController
> >    load_resource only: [:create, :update]
> >
> >    def dashboard
> > -    instantiate_collection { |collection| filter_collection collection }
> > -    respond_with_discussions
> > +    instantiate_collection { |collection| collection_for_dashboard collection }
> > +    respond_with_collection scope: { visible_in_dashboard: true }, serializer: DiscussionWrapperSerializer, root: 'discussion_wrappers'
> > +  end
> > +
> > +  def inbox
> > +    instantiate_collection { |collection| collection_for_inbox collection }
> > +    respond_with_collection scope: { visible_in_inbox: true }, serializer: DiscussionWrapperSerializer, root: 'discussion_wrappers'
>
> So, if you view dashboard, then view inbox, you need to wait for all the
> discussions to be redownloaded with a different variable added to them
> before they'll show?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/loomio/loomio/pull/2369/files#r36117990>.
>